iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/DirectMethodsCommon.java

```diff
@@ -228,6 +244,98 @@ public void openDeviceClientAndSubscribeToMethods() throws Exception
             null);
     }
 
+    public void SubscribeToMethodAndReceiveAsJsonString() throws Exception
+    {
+        testInstance.setup();
+        this.testInstance.identity.getClient().open(true);
+
+        this.testInstance.identity.getClient().subscribeToMethods(
+                (methodName, methodData, context) ->
+                {
+                    String methodDataAsJsonString = methodData.getPayloadAsJsonString();
+
+                    System.out.println("Device invoked " + methodName);
+                    DirectMethodResponse methodResponseAsJsonString;
+                    int status;
+                    Object result;
+
+                    try
+                    {
+                        switch (methodName)
+                        {
+                            case METHOD_MODIFY:
+                                result = modifyPayload(methodDataAsJsonString);
+                                status = UUID.randomUUID().hashCode();
```

Comment:
The use of hashCode here is a bit confusing. How about 
```suggestion
                                status = 1 + new Random().nextInt(998); // random number bound from 1 to 999
```